XamarinCommunityToolkit/Effects/Touch/GestureManager.shared.cs

```diff
@@ -0,0 +1,548 @@
+				using (image.Batch())
+				{
+					image.Aspect = aspect;
+					image.Source = source;
+				}
+			}
+		}
+
+		async Task SetBackgroundColorAsync(TouchEffect sender, TouchState touchState, HoverState hoverState, int duration, Easing easing)
+		{
+			var normalBackgroundColor = sender.NormalBackgroundColor;
+			var pressedBackgroundColor = sender.PressedBackgroundColor;
+			var hoveredBackgroundColor = sender.HoveredBackgroundColor;
+
+			if (normalBackgroundColor == Color.Default &&
+				pressedBackgroundColor == Color.Default &&
+				hoveredBackgroundColor == Color.Default)
+				return;
+
+			var control = sender.Control;
+			if (defaultBackgroundColor == default(Color))
+				defaultBackgroundColor = control.BackgroundColor;
+
+			var color = GetBackgroundColor(normalBackgroundColor);
+
+			if (touchState == TouchState.Pressed)
+				color = GetBackgroundColor(pressedBackgroundColor);
+			else if (hoverState == HoverState.Hovered && sender.Control.IsSet(TouchEffect.HoveredBackgroundColorProperty))
+				color = GetBackgroundColor(hoveredBackgroundColor);
+
+			if (duration <= 0)
+			{
+				control.BackgroundColor = color;
+				return;
+			}
+
+			var animationCompletionSource = new TaskCompletionSource<bool>();
+			new Animation
+            {
+				{ 0, 1,  new Animation(v => control.BackgroundColor = new Color(v, control.BackgroundColor.G, control.BackgroundColor.B, control.BackgroundColor.A), control.BackgroundColor.R, color.R) },
+				{ 0, 1,  new Animation(v => control.BackgroundColor = new Color(control.BackgroundColor.R, v, control.BackgroundColor.B, control.BackgroundColor.A), control.BackgroundColor.G, color.G) },
+				{ 0, 1,  new Animation(v => control.BackgroundColor = new Color(control.BackgroundColor.R, control.BackgroundColor.G, v, control.BackgroundColor.A), control.BackgroundColor.B, color.B) },
+				{ 0, 1,  new Animation(v => control.BackgroundColor = new Color(control.BackgroundColor.R, control.BackgroundColor.G, control.BackgroundColor.B, v), control.BackgroundColor.A, color.A) },
+            }.Commit(sender.Control, nameof(SetBackgroundColorAsync), 16, (uint)duration, easing, (d, b) => animationCompletionSource.SetResult(true));
+			await animationCompletionSource.Task;
+		}
+
+		async Task SetOpacityAsync(TouchEffect sender, TouchState touchState, HoverState hoverState, int duration, Easing easing)
+		{
+			var normalOpacity = sender.NormalOpacity;
+			var pressedOpacity = sender.PressedOpacity;
+			var hoveredOpacity = sender.HoveredOpacity;
+
+			if (Abs(normalOpacity - 1) <= double.Epsilon &&
+				Abs(pressedOpacity - 1) <= double.Epsilon &&
+				Abs(hoveredOpacity - 1) <= double.Epsilon)
+				return;
+
+			var opacity = normalOpacity;
+
+			if (touchState == TouchState.Pressed)
+				opacity = pressedOpacity;
+			else if (hoverState == HoverState.Hovered && sender.Control.IsSet(TouchEffect.HoveredOpacityProperty))
+				opacity = hoveredOpacity;
+
+			await sender.Control.FadeTo(opacity, (uint)Abs(duration), easing);
```

Comment:
```suggestion
			await sender.Control.FadeTo(opacity, (uint)Abs(duration), easing).ConfigureAwait(false);
```

Reply:
I'm wondering if makes sense to warning the dev that `duration` is negative (in some edge case), something like.

```csharp

			if (duration < 0)
				Xamarin.Forms.Internals.Log.Warning("GestureManager", $"Attention! {nameof(duration)} is negative.");
```